lib/open_food_network/products_and_inventory_report_base.rb

```diff
@@ -65,7 +65,15 @@ def filter_to_distributor(variants)
     def filter_to_order_cycle(variants)
       if params[:order_cycle_id].to_i > 0
         order_cycle = OrderCycle.find params[:order_cycle_id]
-        variants.where(id: order_cycle.variants)
+        # There are two quirks here:
+        #
+        # 1. Rails 3 uses only the last `where` clause of a column. So we can't
+        #    use `variants.where(id: order_cycle.variants)` until we upgrade to
+        #    Rails 4.
+        #
+        # 2. `order_cycle.variants` returns an array. So we need to use map
+        #    instead of pluck.
+        variants.where("spree_variants.id in (?)", order_cycle.variants.map(&:id))
```

Comment:
Considering an OC with no variants... The SQL here would end up being `spree_variants.id IN (NULL)`, which is not exactly what we want. But it has the same result (no matches when the selected order cycle has no variants) because `spree_variants.id` here would never be null anyway - the chain starts with `Spree::Variant`. :heavy_check_mark:

Reply:
This adds a performance penalty and we should avoid it given our current situation, particularly with reports. I know `order_cycle.variants` returns an array but this doesn't mean we cannot extract that query and make it reusable.

I've been playing with it and the following patch does pass the tests but issue a single query

```diff
         #
         # 2. `order_cycle.variants` returns an array. So we need to use map
         #    instead of pluck.
-        variants.where("spree_variants.id in (?)", order_cycle.variants.map(&:id))
+        query = Spree::Variant.
+          joins(:exchanges).
+          merge(Exchange.in_order_cycle(order_cycle)).
+          select('DISTINCT spree_variants.*')
+
+        variants.merge(query)
       else
         variants
       end
```

We need to be extra careful with those old model methods that issue queries because it turns out most of them are the root cause of many performance issues :warning: .

Reply:
Pau, my change didn't add a performance penalty. It was there before. I fixed a bug with the least amount of effort and risk while we are very low in resources.

Your suggestion duplicates code but avoids one query. I agree that it's appropriate to have custom queries in reports. It also avoids using the questionable `OrderCycle#variants` which should be refactored.

I added your change as a new commit, @sauloperez, have a look.